### ISSUE TITLE
Need help with local install

### ISSUE THREAD

**Issue by Peter2121** [janis91/ocr#119]
I'm trying to install OCR locally.
I am in FreeBSD 11.1 jail.
Following your wiki, I installed all the necessary components, I have redis up and running, ocrmypdf and tesseract are operational, NodeJS is installed. Now I would like to start worker and configure Nextcloud (currently in another jail, but can be installed in the same one) to communicate with.

In your wiki I found:

> you also have the possibility (mainly because of backwards compatibility reasons), to install the Tesseract, OCRmyPDF, NodeJS (for the worker) and Redis locally and run the node script start like the php worker for version < 3.x. If you want to setup the environment like that you can find the documentation for installation at the corresponding github repositories.

Can you provide more information, please, about:
- starting worker 
- configuring Nextcloud

**Comment by Happyfeet01** [comment]
I have the same question where I find the starter script

**Comment by janis91** [comment]
Basically I prefer the docker setup, because its much easier (from my point of view), but in case you use FreeBSD (or any derivate like FreeNAS or something like that) I guess you are not completely free in choice. The main steps to start and configure the app locally are:

Starting worker:
1. build the worker app: run `npm run build` in the worker directory of the app. This results in a "dist" folder containing the compiled javascript files. This command only has to be used once per new release of the app.
2. start the node-ocr-worker app by running `npm start` in the worker directory of the app. This command can be run as often you want - it is the instance count - for a first try I recommend to start only one instance) please be aware, that this simple command does use the default configuration for redis etc. If you want to configure this (I guess there is a FreeBSD equivalent of doing so), you can start the worker app by adding the environment variables (also set in the docker run statement) in front of the following command, like for example:
`NODE_ENV=production REDIS_PORT=6379 node dist/main.js`

After that you have to configure the Nextcloud jail to be able to connect also to redis. (Redis deals with messages sent to and from nextcloud and the worker) If this is possible, you can configure it in your administration settings like shown in the wiki.

Hope that helps.

**Comment by Peter2121** [comment]
Sorry, but for me it is impossible to build it:
`src/configuration/worker.configuration.ts(2,52): error TS2307: Cannot find module 'child_process'.
src/configuration/worker.configuration.ts(8,13): error TS2304: Cannot find name 'process'.
src/configuration/worker.configuration.ts(9,15): error TS2304: Cannot find name 'process'.
src/configuration/worker.configuration.ts(11,19): error TS2304: Cannot find name 'process'.
src/configuration/worker.configuration.ts(12,15): error TS2304: Cannot find name 'process'.
src/factory/ocr-processing-service.factory.ts(6,27): error TS2307: Cannot find module 'child_process'.
src/factory/ocr-processing-service.factory.ts(10,28): error TS2307: Cannot find module 'fs'.
src/service/ocr-processing.service.ts(9,70): error TS2307: Cannot find module 'child_process'.
src/util/logger.util.ts(34,16): error TS2304: Cannot find name 'process'.
src/worker.ts(50,9): error TS2304: Cannot find name 'process'.
src/worker.ts(53,13): error TS2304: Cannot find name 'process'.
`
I tried to install child_process and fs modules, globally and locally - it does not help.
I am on 8.1.4 version of node.

**Comment by Peter2121** [comment]
I understand that Docker setup is the preferred way of using this application.
But for the moment Docker technology is limited to Linux world. I think that limiting the usage of multi-platform web application to one platform is not a great idea. Hope that this ticket helps to other people to install OCR without Docker.

**Comment by Peter2121** [comment]
After some searching, I uninstalled 'fs' and 'child_process' modules globally and locally, and I could build and start the worker.

**Comment by janis91** [comment]
Is it really working now? Because maybe you need nodejs version 6. I installed and developed the worker with node 6 and npm 3. Sorry I forgot to mention that.

**Comment by Peter2121** [comment]
Globally, it works, but there are still some problems with my install.
1. Starting any new tesseract job, there is an exception in ocr-processing.service.js at this line:
`this.unlinkSync(`${worker_configuration_1.Configuration.OUTPUT_PATH}/${job.tempFile}`);`
It seems to be a bug, as we are trying to delete a new temp file that probably does not exist, and catch "ENOENT: no such file or directory" exception in process(jobMessage).
Commenting this line out, I could pass this step.
2. The result of tesseract job is never passed to Nextcloud - the temporary file contains valid data, but the file, created by Nextcloud in the directory with the initial file, is blank. Probably, I need to adjust some config settings in Nextcloud to use correct temporary directory.

**Comment by janis91** [comment]
Following: You definitely have to configure the temp directory of Nextcloud. This directory has to be the same as the one you linked to / configured for the worker. Then the first point should not throw an error anymore because of the unlinkSync line. This unlinkSync is necessary, as the tesseract command unfortunately doesn't let you specify the full file name that should be created. It always appends "txt" to the given name. Therefore the temp file created by Nextcloud needs to be deleted as we do not need it anymore with tesseract.

TL;DR: This is not a bug. Configure your Nextcloud temp dir properly.

**Comment by Peter2121** [comment]
Yes, you have a reason.
The most important point was that in my config, when Nextcloud temp directory is configured, `Configuration.OUTPUT_PATH` must be blank in `worker.configuration.js`.
So, almost anything works correctly, OCR worker and Nextcloud are installed in two separate jails on FreeBSD server.
I'm going to open another ticket to suggest an enhancement ;)
